internal/activities/validate_sip_name_test.go

```diff
@@ -0,0 +1,178 @@
+package activities_test
+
+import (
+	"fmt"
+	"path/filepath"
+	"testing"
+
+	temporalsdk_activity "go.temporal.io/sdk/activity"
+	temporalsdk_testsuite "go.temporal.io/sdk/testsuite"
+	"gotest.tools/v3/assert"
+	"gotest.tools/v3/fs"
+
+	"github.com/artefactual-sdps/preprocessing-sfa/internal/activities"
+	"github.com/artefactual-sdps/preprocessing-sfa/internal/sip"
+)
+
+func TestValidateSIPName(t *testing.T) {
+	t.Parallel()
+
+	bornDigitalSIPName := "SIP_20010106_someoffice_someref"
+	bornDigitalSIP, err := sip.New(filepath.Join(bornDigitalSIPTempDir(t, bornDigitalSIPName), bornDigitalSIPName))
+	assert.NilError(t, err)
+
+	bornDigitalSIPUnderscoreRefName := "SIP_20010106_someoffice_some_ref"
+	bornDigitalSIPUnderscoreRef, err := sip.New(
+		filepath.Join(bornDigitalSIPTempDir(t, bornDigitalSIPUnderscoreRefName), bornDigitalSIPUnderscoreRefName),
+	)
+	assert.NilError(t, err)
+
+	bornDigitalSIPNoRefName := "SIP_20010106_someoffice"
+	bornDigitalSIPNoRef, err := sip.New(
+		filepath.Join(bornDigitalSIPTempDir(t, bornDigitalSIPNoRefName), bornDigitalSIPNoRefName),
+	)
+	assert.NilError(t, err)
+
+	bornDigitalSIPBadName := "somref"
+	bornDigitalSIPBad, err := sip.New(
+		filepath.Join(bornDigitalSIPTempDir(t, bornDigitalSIPBadName), bornDigitalSIPBadName),
+	)
+	assert.NilError(t, err)
+
+	digitizedSIPName := "SIP_20010106_Vecteur_someref"
```

Comment:
It's not clear from the issue ticket, but I wonder if the DigitizedSIP format requirement of
`SIP_[YYYYMMDD]_Vecteur_[Digitization order ID]` implies that the "Digitization order ID" is a number, rather than an alphanumeric string?

I think you can leave it as is for now, and @sallain or the client can clarify the exact requirements.

Reply:
Sounds good!